### PR TITLE
WeaveTagger and the Overlay topology 

### PR DESCRIPTION
**Pull request by peterbourgon**
PTAL! No changes to the Weave /status-json yet, obviously.

### CODE REVIEW

File: probe/main.go [hunk 1]
```diff
@@ -66,8 +67,12 @@ func main() {
 	}
 	defer publisher.Close()
 
+	var (
+		dockerTagger *tag.DockerTagger
+		weaveTagger  *tag.WeaveTagger
```
Comment (tomwilkie) [inline probe/main.go]: Is tagger the right name?  It produces a topology, doesn't tag an existing one.

**Comment by tomwilkie** [comment]
1 comment, otherwise LGTM.  Hold off until post v0.3.0?

File: probe/main.go (hunk 1 above)
Comment (peterbourgon) [inline probe/main.go]: It is consistent with DockerTagger, which both produces a topology and tags existing topologies. I agree the naming of probe components is currently schizophrenic; that's a consequence of the organic growth of the package, which has resulted in poor separation of concerns and awkward names. I intend to address those things holistically in a refactor...

Comment (tomwilkie) [inline probe/main.go]: SGTM, we've got a ticket for reorganising all this.

File: probe/tag/weave_tagger.go [hunk 2]
```diff
@@ -0,0 +1,102 @@
...
+// OverlayTopology produces an overlay topology from the Weave router.
+func (t WeaveTagger) OverlayTopology() report.Topology {
+	topology := report.NewTopology()
+
+	resp, err := http.Get(t.url)
+	if err != nil {
+		log.Printf("Weave Tagger: %v", err)
+		return topology
+	}
+	defer resp.Body.Close()
```
Comment (tomwilkie) [inline probe/tag/weave_tagger.go]: Note Body.close will block until it reads an EOF.

File: probe/tag/weave_tagger.go [hunk 3]
```diff
@@ -0,0 +1,102 @@
...
+		topology.NodeMetadatas[report.MakeOverlayNodeID(peer.Name)] = report.NodeMetadata{
+			WeavePeerName:     peer.Name,
+			WeavePeerNickName: peer.NickName,
+		}
+	}
+
+	return topology
+}
+
+func sanitize(scheme string, port int, path string) func(string) (string, error) {
```
Comment (tomwilkie) [inline probe/tag/weave_tagger.go]: Why return a function?  As far as I can tell its used immediately.

**Comment by tomwilkie** [comment]
LGTM.  Pls rebase & squash.

File: probe/tag/weave_tagger.go (hunk 3 above)
Comment (peterbourgon) [inline probe/tag/weave_tagger.go]: It's just currying the sanitization parameters. Better separation of concerns than passing params + target string as equal parameters, lighter-weight than defining a method on a POD struct.

**Pull request merged**